tests/integration/test_cogs_rates.py

```diff
@@ -20,6 +20,8 @@ def rates_cog(sample_bot):
             pass
 
 
+# TODO: Update test (and maybe also Rates cog) to test new comparison method
+@pytest.mark.skip(reason="compare_posted_rates method was removed")
```

Comment:
@FM-17 Do you want to take a stab at this maybe? Might not even really be necessary either tbf. 

It might require breaking the `Rates.on_ready()` main loop into some more granular functions to be more testable. Notably, this test is asserting that the `rates_diff` objects are what they should be, but the new approach doesn't currently have any method that returns those diffs. They're also just internal temporary variables rather than instance variables of the cog object. 

Alternatively you could maybe just fetch the rates from the provided URLs using `rates_cog.get_current_rates()` as below, separately compute the diffs, and then keep the same assertions.

That being said, I'd also be happy to leave it alone for now and just circle back later to improve these tests in general. I don't even really see this as a proper integration test anyway, so it's a bit of a misnomer to have it where it is.

Reply:
Just took a stab at this, but I haven't found a path forward that I'm satisfied with.

> It might require breaking the `Rates.on_ready()` main loop into some more granular functions to be more testable.

The main `on_ready()` loop only became more convoluted as I divided it into more granular functions. It flows quite well as is.

> Alternatively you could maybe just fetch the rates from the provided URLs using rates_cog.get_current_rates() as below, separately compute the diffs, and then keep the same assertions. 
> That being said, I'd also be happy to leave it alone for now and just circle back later to improve these tests in general. I don't even really see this as a proper integration test anyway, so it's a bit of a misnomer to have it where it is.

I tried this approach as well, but as you suggested it was a bit of a misnomer to call this an integration test while separately computing components of the test. If something within the main loop were to change, resulting issues wouldn't be caught because the test would remain static. 

I do agree that we should find a suitable replacement for integration testing now that we no longer save rates files. I'd like to leverage the httpserver you set up for this integration test, but I haven't found a clean way of doing it yet.

Reply:
>I'd like to leverage the httpserver you set up for this integration test, but I haven't found a clean way of doing it yet.

I can walk you through the process of using the `httpserver` fixture to be the "input" side of the integration test, but we'll need something else to use on the "output" side. I'm thinking [dpytest](https://github.com/CraftSpider/dpytest/blob/master/tests/test_verify_embed.py#L6-L18).

The question is whether it needs to be done before proceeding with this PR, or whether we can find another way to manualy test sufficiently for now and then circle back to testing afterwards. Either way though, it should be in a PR of its own I think though. This one can just be put on temporary hold if you prefer the former option.

Reply:
>The main on_ready() loop only became more convoluted as I divided it into more granular functions. It flows quite well as is.

Come to think of it though, I just realized that the core functionality is fundamentally written inside of a `while True` loop, so if we wanted to test the bot's main functionality, I think it may have to be broken out into a single function that performs the action exactly once. Something that we can execute from within `pytest` a known number of times.